### PR TITLE
Adds "Strong Stomach" quirk, a core CDDA/PZ quirk we've sorely been missing. Also Deviant Tastes dirty food re-nerf.

### PR DESCRIPTION
**Pull request by MrMelbert**
## About The Pull Request

- Adds Strong Stomach quirk. 
   - 4 points
   - You can eat dirty food without risk of getting disease. 
   - You suffer less negative effects from vomiting. Vomit stuns you for half the duration, and you lose half as much nutrition. 

- Reverts https://github.com/tgstation/tgstation/pull/76864 , integrates its effects into Strong Stomach instead. 

## Why It's Good For The Game

- Lotta people (namely Lizards and sometimes Felines with Deviant Tastes) run gimmicks involving them being a gremlin person and eating trash off the ground, and it's rather hard to accomplish this now since it makes you a public medbay enemy # 1. This quirk should give them an option to avoid that. 
   - Also (as mentioned in the title) both CDDA and PZ have this trait and I can't believe we're missing it! This is something in modifiable-character-traits/quirks-101. 

- I moved the effects from #76864 to this quirk because 1. I thought it was more fitting and 2. I thought the original PR was kinda wack for what is (generally) a neutral quirk. 

## Changelog

:cl: Melbert
add: Adds the Strong Stomach quirk, which allows you to eat grimy food without worry about disease, and makes you a bit more resilient to the effects of vomiting. 
del: Deviant Tastes no longer prevents you from getting a negative moodlet from eating dirty food. Strong Stomach does that now. 
/:cl:

### CODE REVIEW

**Comment by jughu** [comment]
rust heretic buff i think

**Comment by DaCoolBoss** [comment]
Attach the trait to the stomach so it can potentially be lost/stolen.

**Comment by carlarctg** [comment]
this trait crosses the line from 'fun extra gimmick for your character' into 'qol so useful and vital its a must-have'.

**Comment by MercuryArrow** [comment]
> this trait crosses the line from 'fun extra gimmick for your character' into 'qol so useful and vital its a must-have'.

I dunno if I'd go that far - vomiting's stun is very annoying but doing so isn't terribly common outside of heavy toxin exposure, spinning on purpose or certain diseases. 
Speaking of which, of the three floor diseases, the only one with vomiting as a symptom is borderline harmless, one is only particularly dangerous if multiple people have it and remain together in an enclosed space with no scrubbers, and the carp one is probably most annoying but is largely unchanged by this. I've seen people cultivate all three of these on purpose for gimmicks before, especially back when they used to be customisable via PANDEMIC, so losing the primary way to access them actually hurts some maint gremlin gimmicks too.

Overall, it feels fine at a 4 pointer - tge benefits are good but they're situational for most people.

**Comment by Loafin34** [comment]
add this but make sure that if they bleed from the chest they will leak stomach acid, aka hydrochloric acid, which causes minor burn damage. reason being that the strong stomach has better stomach filtering but worse stomach lining

File: code/datums/quirks/positive_quirks/strong_stomach.dm [hunk 1]
```diff
@@ -0,0 +1,12 @@
+/datum/quirk/strong_stomach
+	name = "Strong Stomach"
+	desc = "You can eat food discarded on the ground without getting sick, and vommiting affects you less."
```
Comment (Jacquerel) [inline code/datums/quirks/positive_quirks/strong_stomach.dm:3]:
```suggestion
	desc = "You can eat food discarded on the ground without getting sick, and vomiting affects you less."
```

**Pull request merged**